### PR TITLE
login-logout-remember

### PR DESCRIPTION
**Pull request by phuong1492**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class UsersController < ApplicationController
+  before_action :logged_in_user, only: [:show, :update]
+
+  def show
+    @user = User.find(params[:id])
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: remove ()

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+class SessionsController < ApplicationController
+  def new
+  end
+
+  def create
+    user = User.find_by(email: params[:session][:email].downcase)
```
Comment (dieunb) [inline app/controllers/sessions_controller.rb]: remove ()

File: app/assets/javascripts/sessions.coffee [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (chuanhtuan) [inline app/assets/javascripts/sessions.coffee]: remove file

File: app/assets/javascripts/users.coffee [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (chuanhtuan) [inline app/assets/javascripts/users.coffee]: remove file

File: app/assets/stylesheets/sessions.scss [hunk 5]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Sessions controller here.
```
Comment (chuanhtuan) [inline app/assets/stylesheets/sessions.scss]: remove file

File: app/assets/stylesheets/users.scss [hunk 6]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Users controller here.
```
Comment (chuanhtuan) [inline app/assets/stylesheets/users.scss]: remove file

File: app/controllers/sessions_controller.rb [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    if user && user.authenticate(params[:session][:password])
+      # Log the user in and redirect to the user's show page.
+      log_in user
+      params[:session][:remember_me] == '1' ? remember(user) : forget(user)
+      redirect_to user
+    else
+      # Create an error message.
+      flash.now[:danger] = 'Invalid email/password combination'
```
Comment (chuanhtuan) [inline app/controllers/sessions_controller.rb]: use " instead of '

File: app/models/user.rb [hunk 8]
```diff
@@ -1,2 +1,40 @@
...
+  before_save { self.email = email.downcase }
+  validates :name,  presence: true, length: { maximum: 50 }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: { maximum: 255 },
+                    format: { with: VALID_EMAIL_REGEX },
+                    uniqueness: { case_sensitive: false }
+  has_secure_password
+  validates :password, length: { minimum: 6 },allow_blank: true
+  
+  # Returns the hash digest of the given string.
```
Comment (chuanhtuan) [inline app/models/user.rb]: remove all remark code

File: app/models/user.rb [hunk 9]
```diff
@@ -1,2 +1,40 @@
...
+
+  # Returns a random token.
+  def User.new_token
+    SecureRandom.urlsafe_base64
+  end
+
+  # Remembers a user in the database for use in persistent sessions.
+  def remember
+    self.remember_token = User.new_token
+    update_attributes(:remember_digest => User.digest(remember_token))
```
Comment (chuanhtuan) [inline app/models/user.rb]: update_attributes(:remember_digest: User.digest(remember_token))

File: app/models/user.rb [hunk 10]
```diff
@@ -1,2 +1,40 @@
...
+
+  # Returns true if the given token matches the digest.
+  def authenticated?(remember_token)
+    return false if remember_digest.nil?
+    BCrypt::Password.new(remember_digest).is_password?(remember_token)
+  end
+
+   # Forgets a user.
+  def forget
+    update_attribute(:remember_digest, nil)
```
Comment (chuanhtuan) [inline app/models/user.rb]: use update_attributes instead of update_attribute

File: app/views/sessions/new.html.erb [hunk 11]
```diff
@@ -0,0 +1,23 @@
+<% provide(:title, "Log in") %>
+<h1>Log in</h1>
+
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for(:session, url: login_path) do |f| %>
+
```
Comment (chuanhtuan) [inline app/views/sessions/new.html.erb]: error message???

**Pull request closed without merging**